### PR TITLE
added check for no objects detected in object detector

### LINKED ISSUE #435: Behavior of fastrcnn object_detector demo with a black image and no predictions
**Setup**
Running fastrcnn object_detector [demo](https://github.com/pytorch/serve/tree/master/examples/object_detector/fast-rcnn), without modifications, on a JPEG image that is completely black.  No objects are detected (as expected).

**Summary of Issue**
An exception occurs based in object_detector.py, in inference(), at the line

```pred_t = [pred_score.index(x) for x in pred_score if x > threshold][-1]```

because pred_score is an empty list.  The exception in the logs is:
```raise Exception("Please provide a custom handler in the model archive.")```

and the JSON returned is

```
{
  "code": 503,
  "type": "InternalServerException",
  "message": "Prediction failed"
}
```

**Expected Behavior**
1) Not a "Prediction failed" indication, that can't be differentiated from other scenarios.  Possibly "no predictions found".
2) Possibly a configurable option on what to do for an empty prediction list.  I've currently worked around this by returning a length-1 of `__background__`, but this may not be the best solution for all models, pre-trained or otherwise.

### PR DESCRIPTION
**Pull request by harshbafna**
## Description

Fixed object detector default handler for "no objects detected" by the model.

Fixes #435 

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## Feature/Issue validation/testing

 - Start torchserve
 - register fastrcnn model from model zoo
 - run inference with a [blank image](https://user-images.githubusercontent.com/26594595/84241556-3fbd6700-ab1d-11ea-9d6b-1e874f318c54.jpg)
 - The request should succeed with response "No object detected"

## Checklist:

- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [x] New and existing unit tests pass locally with these changes?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: ts/torch_handler/object_detector.py [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
 import io
+import logging
 import torch
 from PIL import Image
 from torchvision import transforms
 from torchvision import __version__ as torchvision_version
 from torch.autograd import Variable
 from .vision_handler import VisionHandler
 
+logger = logging.getLogger(__name__)
```
Comment (dhaniram-kshirsagar) [inline ts/torch_handler/object_detector.py:13]: Remove unused object.

File: ts/torch_handler/object_detector.py [hunk 2]
```diff
@@ -91,8 +97,9 @@ def handle(data, context):
 
         data = _service.preprocess(data)
         data = _service.inference(data)
-        data = _service.postprocess(data)
-
-        return data
+        if data:
+            return _service.postprocess(data)
+        else:
+            return ["No object detected"]
```
Comment (dhanainme) [inline ts/torch_handler/object_detector.py:103]: Wondering if returning an empty list is more appropriate that returning an element a single element with string val.

**Comment by dhanainme** [comment]
Can you also add tests that exercise this code path to confirm that it works.

**Comment by harshbafna** [comment]
> Can you also add tests that exercise this code path to confirm that it works.

Added a test case in the sanity script with a blank image.

File: ts/torch_handler/object_detector.py (hunk 2 above)
Comment (harshbafna) [reply ts/torch_handler/object_detector.py:103]: I am open to both ideas. Can remove the if-else block if you want .

Comment (maaquib) [reply ts/torch_handler/object_detector.py:103]: Agreed with @dhanainme

Comment (harshbafna) [reply ts/torch_handler/object_detector.py]: Done.

**Pull request merged**